WordPress/Classes/ViewRelated/Reader/ReaderBaseHeaderView.swift

```diff
@@ -0,0 +1,46 @@
+import UIKit
+import WordPressUI
+
+// A container view for stream headers.
+class ReaderBaseHeaderView: UIView {
+    let contentView = UIView()
+
+    var isCompact: Bool = true {
+        didSet {
+            guard oldValue != isCompact else { return }
+            setNeedsUpdateConstraints()
+        }
+    }
+
+    private var contentViewConstraints: [NSLayoutConstraint] = []
+
+    override init(frame: CGRect) {
+        super.init(frame: frame)
+
+        addSubview(contentView)
+    }
+
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    override func updateConstraints() {
+        NSLayoutConstraint.deactivate(contentViewConstraints)
+        contentViewConstraints = []
+
+        let insets = Self.makeInsets(isCompact: isCompact)
+        contentViewConstraints += contentView.pinEdges(.horizontal, to: isCompact ? self : readableContentGuide, insets: insets)
+        contentViewConstraints += contentView.pinEdges(.vertical, insets: insets)
```

Comment:
Nitpick: I believe the same thing can be implemented using `layoutMarginGuide`, which doesn't need a new "content view".